string[] name
float64[] probability
float64[] top
float64[] bottom
float64[] left
float64[] right